src/extras/Earcut.js

```diff
@@ -362,17 +374,20 @@ function compareX( a, b ) {
 // find a bridge between vertices that connects hole with an outer ring and link it
 function eliminateHole( hole, outerNode ) {
 
-	outerNode = findHoleBridge( hole, outerNode );
-	if ( outerNode ) {
+	const bridge = findHoleBridge( hole, outerNode );
 
-		const b = splitPolygon( outerNode, hole );
+	if ( ! bridge ) {
 
-		// filter collinear points around the cuts
-		filterPoints( outerNode, outerNode.next );
-		filterPoints( b, b.next );
+		return outerNode;
 
 	}
 
+	const bridgeReverse = splitPolygon( bridge, hole );
+
+	// filter collinear points around the cuts
+	return filterPoints( bridgeReverse, bridgeReverse.next ); // there is different from the implementation of Earcut v2.2.4
```

Comment:
Why did you change the earcut implementation here? If there is a bug, wouldn't it be better to report it to the Earcut repository instead?

Reply:
To clarify my initial post: A broken E2E test potentially indicates a bug. That's why I have asked to verify why the output has changed. However, if the output is as intended (because something has been fixed/improved in the earcut code), it's not necessary to restore the previous behavior. In this case, it's fine to update the screenshot.

It's just important to know, why the output has changed.

Reply:
I tagged here for assistance because there might be a problem (but it still returned the triangulation when I tested Earcut).

I'm feel so sorry about I did misunderstand what you meant yesterday.

I encountered an infinite loop problem in a particular model when I was learning the 3D engine, so I wanted to try updating the referenced earcut library to solve the problem. I'll reread the code until I know what's going on. Thank you for your review.

Reply:
> I encountered an infinite loop problem in a particular model when I was learning the 3D engine

If you have the time it might be interesting to demonstrate the issue with a live example: https://jsfiddle.net/f2Lommf5/